src/main/java/hudson/plugins/git/util/GitUtils.java

```diff
@@ -190,7 +219,11 @@ public Revision sortBranchesForRevision(Revision revision, List<BranchSpec> bran
                         }
 
                     } finally {
-                        walk.release();
+                        try {
+                            walk.release();
```

Comment:
I don't think this code path is heavy enough (or long enough lived) for the added sophistication of the lambda.